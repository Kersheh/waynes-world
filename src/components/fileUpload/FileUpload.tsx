import React, { useRef } from 'react';
import { useDispatch } from 'react-redux';
import imageCompression from 'browser-image-compression';

import Button from 'components/button/Button';
import { setEditAlbumArtAction } from 'containers/addAlbum/addAlbumActions';
import styles from './FileUpload.module.scss';

interface FileUploadProps {
  children: React.ReactNode;
}
const FileUpload = ({ children }: FileUploadProps) => {
  const dispatch = useDispatch();
  const fileInputRef = useRef<HTMLInputElement>(null);

  return (
    <div className={styles.fileUpload}>
      <Button styleType="basic" onClick={() => fileInputRef?.current?.click()}>
        <input
          type="file"
          ref={fileInputRef}
          onChange={async e => {
            if (e.target.files && e.target.files.length > 0) {
              // compress image down to 300px
              const compressedImg = await imageCompression(e.target.files[0], {
                maxWidthOrHeight: 300
              });

              // convert image to base64 and update active edit album art
              const fileReader = new FileReader();
              fileReader.readAsDataURL(compressedImg);
              fileReader.onloadend = ({ target }) => {
                dispatch(
                  setEditAlbumArtAction({
                    base64: target?.result ?? null
                  })
                );
              };
            }
          }}
        />

        {children}
      </Button>
    </div>
  );
};

export default FileUpload;
