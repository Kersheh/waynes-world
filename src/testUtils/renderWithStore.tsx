import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose, Reducer } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { render } from '@testing-library/react';

import rootReducer from 'store/reducers';
import rootSaga from 'store/sagas';

const renderWithStore = (component: React.ReactElement) => {
  const sagaMiddleware = createSagaMiddleware();
  const store = createStore(
    rootReducer as Reducer,
    compose(applyMiddleware(sagaMiddleware))
  );
  sagaMiddleware.run(rootSaga);

  return render(<Provider store={store}>{component}</Provider>);
};

export default renderWithStore;
