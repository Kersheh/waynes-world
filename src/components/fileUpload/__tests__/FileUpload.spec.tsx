import React from 'react';

import renderWithStore from 'testUtils/renderWithStore';
import FileUpload from '../FileUpload';

describe('<FileUpload/> component', () => {
  it('should render', () => {
    const { container } = renderWithStore(
      <FileUpload>
        <div />
      </FileUpload>
    );

    expect(container).toMatchSnapshot();
  });
});
