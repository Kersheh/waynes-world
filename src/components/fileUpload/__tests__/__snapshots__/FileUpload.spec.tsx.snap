// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileUpload/> component should render 1`] = `
<div>
  <div
    class="fileUpload"
  >
    <button
      class="button basic"
      type="button"
    >
      <input
        type="file"
      />
      <div />
    </button>
  </div>
</div>
`;
